blazingcache-core/src/main/java/blazingcache/network/jvm/JVMChannel.java

```diff
@@ -44,13 +45,15 @@
 public class JVMChannel extends Channel {
 
     private static final Logger LOGGER = Logger.getLogger(JVMChannel.class.getName());
+    private static final AtomicLong ID_GENERATOR = new AtomicLong();

```

Comment:
why are you using a "static" variable ?
this smells to me because we have two cases (I didn't check the wire protocol):
1) this ID is unique in the scope of the connection (channel?), then it is better to use an instance field
2) this ID is unique globally......then having a static field is a bad way to generate this value, because every JVM will have its own counter


I guess we are in case 2), so I would use a simple instance field

Reply:
The protocol works like this:

- jvm1 sends a request_message with messageID=x
- jvm1 stores the entries <messageID, callback> <messageID, request_message>, this are stored in the channel
- jvm2 receives request_message and sends a reply_message with replyMessageID=request_message.messageID and messageID=k
- jvm1 receives the reply_message and reads the replyMessageID to invoke the right callback and stuff

Even if the request_message.messageID is stored at channel level, the whole request_message gets passed to callbacks which comes from outside the channel (eg. see CacheClient#touchEntry). So I think it's good to have request_messages' messageID unique across the jvm channels

The reply messageID can conflict with the messageID of the request, but reply_message.messageID is never used. This can also happen with uuid v4 (with a very lower chance)

Reply:
the messageId is unique only in the scope of a "Channel" and actually you have two series of ids, once per each side of the connection.

even if the message id is leaked in callback, it is a useless information for the caller

I think that having a "static" id generator is only misleading.

I am trying to find a valid usecase, what would you do with a unique id per-JVM (classloader) ? logging the message id and correlate errors ?

Reply:
Yes logging and tracking the message. With a local message id we would have to expose and log the channel id too.

Right now there are no such logs, It was just a thought for future. I will make the generator local